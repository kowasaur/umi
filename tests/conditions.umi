Void main() {
    mut String variable = "unmodified"
    if true {
        variable = "modified"
        print("this should print")
    }

    if 1 == 1 && (2 == 2) {
        String variable = "this is block scoped"
        print(variable)
    }

    print(variable)

    if variable == "modified" { print("this should also print") }
    if variable == "unmodified" { print("this shouldn't print") }

    if 5 > 123 {
        print("maths is broken")
    } else if 3 < 10000 {
        print("maths makes sense")
    } else {
        print("maths is broken")
    }

    Int number = if variable != "unmodified" {
        Int a = 3 * 4
        a * 5 + 9
    } else {
        print("this will not print")
        0
    }
    print(if number >= 69 { "wow you can use if statements as values" } else { ":(" })
}
