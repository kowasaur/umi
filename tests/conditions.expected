this should print
this is block scoped
modified
this should also print
maths makes sense
wow you can use if statements as values
